x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilegeTests.java

```diff
@@ -68,11 +67,7 @@ public void testFindPrivilegesThatGrant() {
         assertThat(findPrivilegesThatGrant(DeleteAction.NAME), equalTo(List.of("delete", "write", "all")));
         assertThat(
             findPrivilegesThatGrant(IndicesStatsAction.NAME),
-            equalTo(
-                Stream.of("monitor", (TcpTransport.isUntrustedRemoteClusterEnabled() ? "cross_cluster_replication" : null), "manage", "all")
```

Comment:
I intentionally kept the `nonNull` filter in `ClusterPrivilegesResolver` since it could be handy in future and we still need the filter for places like `IndexPrivilege` and `Constants`. But the usage here and maybe other places should be removed since its pure clutter. I'll do another sweep. Thanks!